src/ray/core_worker/reference_count.h

```diff
@@ -60,20 +77,28 @@ class ReferenceCounter {
   /// possible to have leftover references after a task has finished.
   ///
   /// \param[in] object_id The ID of the object that we own.
+  /// \param[in] inner_ids ObjectIDs that are contained in the object's value.
+  /// As long as object_id is in scope, the inner objects should not be GC'ed.
   /// \param[in] owner_id The ID of the object's owner.
   /// \param[in] owner_address The address of the object's owner.
   /// \param[in] dependencies The objects that the object depends on.
-  void AddOwnedObject(const ObjectID &object_id, const TaskID &owner_id,
-                      const rpc::Address &owner_address) LOCKS_EXCLUDED(mutex_);
+  void AddOwnedObject(const ObjectID &object_id, const std::vector<ObjectID> &inner_ids,
```

Comment:
```suggestion
  void AddOwnedObject(const ObjectID &object_id, const std::vector<ObjectID> &contained_ids,
```
I think `contained` is more clear unless you're going to call `object_id` the `outer_id`